barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.hpp

```diff
@@ -468,6 +478,9 @@ void handle_memory_op(Program::Opcode::MemoryOp const& mem_op, BlockConstraint&
     if (block.type == BlockType::ROM && access_type == 1) {
         block.type = BlockType::RAM;
     }
+    if (access_type == 1) {
```

Comment:
I can see that this is confusing.
The idea is that the block type is defined here by BB, depending on the operations.
It was set by default to ROM and then when there is a write, it is switched it to RAM.
Now that we also have databus type, I wanted to separate the logic so that this is only valid for RAM or ROM.
I have re-factored this code so that it is simpler, and also added a comment.